src/Application/Commands/PushDailyFundTotals.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MatchBot\Application\Commands;
+
+use MatchBot\Application\Messenger\FundTotalUpdated;
+use MatchBot\Domain\FundRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\RoutableMessageBus;
+
+/**
+ * @see RetrospectivelyMatch which does a similar thing for just-closed campaigns as one of the last campaign close tasks.
+ */
+#[AsCommand(
+    name: 'matchbot:push-daily-fund-totals',
+    description: 'Pushes champion funds used totals to Salesforce, for all associated with open campaigns'
+)]
+class PushDailyFundTotals extends LockingCommand
+{
+    public function __construct(
+        private readonly FundRepository $fundRepository,
+        private readonly LoggerInterface $logger,
+        private readonly RoutableMessageBus $bus,
+    ) {
+        parent::__construct();
+    }
+
+    public function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $funds = $this->fundRepository->findForCampaignsOpenAt(new \DateTime('now'));
```

Comment:
```suggestion
        $funds = $this->fundRepository->findForCampaignsOpenAt(new \DateTimeImmutable('now'));
```

(will need the matching change in repository). Worth treating DateTime as deprecated IMHO, especially since it somewhat dangerously has methods with the same names as the DateTimeImmutable that mutate it, so if we don't remember which one we have we could mutate by mistake.

I'd also inject the timestamp via the constructor but unless you're going to write a test against this class I don't think that's so important.

Reply:
Type change done in https://github.com/thebiggive/matchbot/pull/1310/commits/5fef1730299dba20e1d8b21c61d282a50260aed2